src/abstractblocksparsearray/abstractblocksparsematrix.jl

```diff
@@ -1 +1,22 @@
 const AbstractBlockSparseMatrix{T} = AbstractBlockSparseArray{T,2}
+
+# SVD is implemented by trying to
+# 1. Attempt to find a block-diagonal implementation by permuting
+# 2. Fallback to AbstractBlockArray implementation via BlockedArray
+function svd(
+  A::AbstractBlockSparseMatrix; full::Bool=false, alg::Algorithm=default_svd_alg(A)
+)
+  T = LinearAlgebra.eigtype(eltype(A))
+  A′ = try_to_blockdiagonal(A)
```

Comment:
I found it confusing that this is call `A′` but if it is a blocked permutation it is actually a composite object of the block diagonal matrix and the block permutations, we should make that clearer in the variable name, i.e. `A′_and_blockperms`.